content/combinatorial/chapter.tex

```diff
@@ -65,6 +80,11 @@ \section{General purpose numbers}
 		$1, 1, 2, 5, 15, 52, 203, 877, 4140, 21147, \dots$. For $p$ prime,
 		\[ B(p^m+n)\equiv mB(n)+B(n+1) \pmod{p} \]
 
+	\subsection{Labeled unrooted trees}
+		\# on $n$ vertices: $n^{n-2}$ \\
+		\# on $k$ existing trees of size $n_i$: $n_1n_2\cdots n_k n^{k-2}$ \\
+		\# with degrees $d_i$: $(n-2)! / ((d_1-1)! \cdots (d_n-1)!)$
```

Comment:
I don't get this part. If every node has degree 1 this formula says the number of trees should be `(n-2)!`, but it's actually 0?

Reply:
hmm, yeah, it doesn't seem to be quite correct, looking here: https://cp-algorithms.com/graph/pruefer_code.html#toc-tgt-7

Seems like the correct formula is ![](https://i.imgur.com/wcSvM18.jpg), where `s_i` is the size of the i-th connected component, where we have `k` components to begin with.

Reply:
Apparently I think I just missed the silent assumption that the sum of the degrees must be 2k - 2 for the formula to hold.